common/os_calls.c

```diff
@@ -154,6 +154,36 @@ g_init(const char *app_name)
 
     WSAStartup(2, &wsadata);
 #endif
+#if defined(XRDP_NVENC)
+    if (g_strcmp(app_name, "xrdp") == 0)
+    {
+        /* call cuInit() to initalize the nvidia drivers */
+        /* TODO create an issue on nvidia forums to figure out why we need to
+        *  do this */
+        if (g_fork() == 0)
+        {
+            typedef int (*cu_init_proc)(int flags);
+            cu_init_proc cu_init;
+            long lib;
+            char cuda_lib_name[] = "libcuda.so";
+            char cuda_func_name[] = "cuInit";
+
+            lib = g_load_library(cuda_lib_name);
+            if (lib != 0)
+            {
+                cu_init = (cu_init_proc)
+                          g_get_proc_address(lib, cuda_func_name);
+                if (cu_init != NULL)
+                {
+                    cu_init(0);
+                }
+            }
+            log_end();
+            g_deinit();
+            g_exit(0);
+        }
+    }
```

Comment:
I don't t think the log ing system is initialized at this point.  Can we still log?

Reply:
You're quite right. Calling `LOG()` won't work.

Two possible solutions:-
1) Log errors to `stderr` in this function. This isn't ideal, but it will at least get picked up by the journal on systemd-based systems.
2) Use a separate function for this purpose (`g_cu_init()`?) called explicitly from sesman after logging has started. We can remove it if you ever get a better answer as to why this function is required.